src/app/modules/bounce/bounce.service.ts

```diff
@@ -235,7 +235,9 @@ export const notifyAdminsOfBounce = async (
       recipientEmail: recipient.email,
     }),
   )
-  const smsResults = await Promise.allSettled(smsPromises)
+
+  // neverthrow#combine is not used since we do not want to short circuit on the first error.
+  const smsResults = await Promise.all(smsPromises)
```

Comment:
> neverthrow#combine is not used since we do not want to short circuit on the first error.

Wouldn't `Promise.all` also lose type safety, and short circuit upon the first error encountered? Why not stick with `Promise.allSettled`?

Reply:
nope, because now we're awaiting on an array of `ResultAsync` instead of `Promise`, so all of them will resolve

Reply:
Are you sure? I'm tracing the code and `smsPromises` comes from `BounceService.getEditorsWithContactNumbers`, which has these line of code, so it looks like it is returning a `Promise<UserWithContactNumber[]>` that never fails?

```typescript
if (smsRecipientsResult.isOk()) {
    return smsRecipientsResult.value.filter(
      (r) => !!r.contact,
    ) as UserWithContactNumber[]
  } else {
    ...
    return []
  }
```

1. We should be using a type guard to filter the array, instead of typecasting to `UserWithContactNumber[]`.
2. I noticed `BoundService.findContactsForEmails` looks just like `UserService.getEditorsWithContactNumbers`, except it performs a filter for the `contact` property. I think we could have simply extended the API instead of creating another service function. It's less common but still OK to cross-call services from other modules.

Reply:
I'm not sure you're tracing correctly - there are two instances of `await Promise.all(smsPromises)`, one comes from `SmsFactory.sendFormDeactivatedSms` and one comes from `SmsFactory.sendBouncedSubmissionSms`, both of which return a `ResultAsync`. so the type of `smsPromises` is always `ResultAsync[]`. you can check out the branch and hover over to see the type of `smsPromises` if you want to verify this.

I agree with your points 1 and 2. we cross-call services from other modules all the time, no qualms about doing that.